File: src/PrimaRoleReactions/db_access/src/role_reaction_info.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct RoleReactionInfo {
    pub guild_id: String,
    pub channel_id: String,
    pub emoji_id: String,
    pub role_id: String,
    pub eureka: Option<bool>,
    pub bozja: Option<bool>,
}

File: src/Prima.Data/RoleReactionInfo.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Prima.Data;

public class RoleReactionInfo
{
    [BsonId]
    [BsonRequired]
    [BsonElement("_id")]
    public ObjectId Id { get; set; }

    [BsonElement("guild_id")] public string? GuildId { get; set; }

    [BsonElement("channel_id")] public string? ChannelId { get; set; }

    [BsonElement("emoji_id")] public string? EmojiId { get; set; }

    [BsonElement("role_id")] public string? RoleId { get; set; }

    [BsonElement("eureka")] public bool? Eureka { get; set; }

    [BsonElement("bozja")] public bool? Bozja { get; set; }
}